crypto/digestible/signature/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+#![no_std]
+
+use mc_crypto_digestible::{Digestible, MerlinTranscript};
+use signature::{Signature, Signer, Verifier};
+
+pub use signature::Error;
+
+/// Construct a deterministic semantic signature over a given object.
+pub trait DigestibleSigner<S: Signature, T: Digestible> {
+    fn sign_semantic(&self, context: &'static [u8], message: &T) -> S;
```

Comment:
Why require a static lifetime? My understanding is that you could still pass an argument with a static lifetime to a function that accepts any lifetime, but requiring a static lifetime limits the inputs to only accepting ones with a static lifetime. Is there a benefit or a reason this is required?

Reply:
The `merlin::Transcript::new()` signature imposes this. Ultimately it doesn't matter, because those context strings probably *should* be hard-coded.